File: connectors/connector-trello/src/test/scala/com/taskadapter/connector/trello/TrelloFactoryTest.scala
package com.taskadapter.connector.trello

import com.taskadapter.model.AllFields
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class TrelloFactoryTest extends FunSpec with Matchers {
  val defaultFieldsForNewConfig = new TrelloFactory().getDefaultFieldsForNewConfig

  describe("default fields") {
    it("have Trello list name") {
      defaultFieldsForNewConfig should contain(TrelloField.listName)
    }
    it("do not have these") {
      defaultFieldsForNewConfig should contain noneOf(TrelloField.listId, AllFields.reporterFullName,
        AllFields.reporterLoginName, AllFields.id, AllFields.key)
    }
  }
}

File: connectors/connector-trello/src/test/java/com/taskadapter/connector/trello/TrelloFactoryTest.java
package com.taskadapter.connector.trello;

import com.taskadapter.model.AllFields;
import com.taskadapter.model.Field;
import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class TrelloFactoryTest {
    private static final List<Field<?>> defaultFieldsForNewConfig = new TrelloFactory().getDefaultFieldsForNewConfig();

    @Test
    public void defaultFieldsHaveTrelloListName() {
        assertThat(defaultFieldsForNewConfig).contains(TrelloField.listName);
    }

    @Test
    public void defaultFieldsDoNotHaveThese() {
        assertThat(defaultFieldsForNewConfig).doesNotContain(TrelloField.listId, AllFields.reporterFullName,
                AllFields.reporterLoginName, AllFields.id, AllFields.key);
    }
}
